src/lib/y2storage/guided_proposal.rb

```diff
@@ -205,17 +205,58 @@ def candidate_devices_with_empty_partition_table(devicegraph)
       devices.select { |d| d.partition_table && d.partitions.empty? }
     end
 
+    # Returns the target sizes to make the proposal
+    #
+    # @return [Array<Symbol>]
+    def target_sizes
+      [:desired, :min]
+    end
+
     # Copy of the original settings including some calculated and necessary
     # values (mainly candidate_devices), in case they were not present
     #
     # @return [ProposalSettings]
     def populated_settings
-      return @populated_settings if @populated_settings
+      @populated_settings ||= settings.dup
+    end
 
-      populated = settings.dup
-      populated.candidate_devices ||= disk_analyzer.candidate_disks.map(&:name)
+    # List of candidate devices
+    #
+    # If they are not given in the settings, disks found by the analyzer will be used.
+    #
+    # @see #candidate_devices_from_settings
+    # @see #candidate_devices_from_analyzer
+    #
+    # @return [<Array<Array<String>>]
+    def candidate_devices
+      return [settings.candidate_devices] if settings.candidate_devices
+      @candidate_devices ||= candidate_devices_from_settings || candidate_devices_from_analyzer
+    end
 
-      @populated_settings = populated
+    # List of candidate devices given in settings
+    #
+    # @return [<Array<Array<String>>]
+    def candidate_devices_from_settings
+      [settings.candidate_devices] if settings.candidate_devices
+    end
+
+    # List of candidate devices based on disks available in the analyzer
+    #
+    # By now, it consist on a sequence of available disk plus all of them at the end (unless
+    # there is only one, to avoid duplications). E.g, if available disks are
+    #
+    # ["/dev/sda", "/dev/sdb"]
+    #
+    # candidates list will be
+    #
+    # [["/dev/sda"], ["/dev/sdb"], ["/dev/sda", "/dev/sdb"]]
+    #
+    # @return [<Array<Array<String>>]
+    def candidate_devices_from_analyzer
+      disk_names = disk_analyzer.candidate_disks.map(&:name)
+      candidates = disk_names.zip
+      candidates << disk_names unless candidates.include?(disk_names)
+      candidates
```

Comment:
NP: why not this version?

```suggestion
      disk_names = disk_analyzer.candidate_disks.map(&:name)
      disk_names.zip.append(disk_names).uniq
```

Reply:
Because two main reasons:

* I did not think in this exact version :P
* I had a similar line of code but I though that could be a bit more complicate to understand.

However, I'll apply your suggestion.

Thanks!